ngraph/core/shape_inference/include/experimental_detectron_prior_grid_generator_shape_inference.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/experimental_detectron_prior_grid_generator.hpp>
+
+namespace ov {
+namespace op {
+namespace v6 {
+
+template <class T>
+void shape_infer(const ExperimentalDetectronPriorGridGenerator* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 3 && output_shapes.size() == 1);
+    const auto& priors_shape = input_shapes[0];
+    const auto& featmap_shape = input_shapes[1];
+    const auto& im_data_shape = input_shapes[2];
+
+    auto& output_shape = output_shapes[0];
+    size_t output_size = op->m_attrs.flatten ? 2 : 4;
+
+    output_shape.resize(output_size);
+    output_shape[output_size - 1] = 4;
+
+    if (priors_shape.rank().is_dynamic() || featmap_shape.rank().is_dynamic()) {
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(op, priors_shape.rank().compatible(2), "Priors rank must be equal to 2.");
+
+    if (priors_shape[1].is_static()) {
+        NODE_VALIDATION_CHECK(op,
+                              priors_shape[1].compatible(4),
+                              "The last dimension of the 'priors' input must be equal to 4. Got: ",
+                              priors_shape[1]);
+    }
+
+    NODE_VALIDATION_CHECK(op, featmap_shape.rank().compatible(4), "Feature_map rank must be equal to 4.");
+
+    if (im_data_shape.rank().is_dynamic()) {
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(op, im_data_shape.rank().compatible(4), "Im_data rank must be equal to 4.");
```

Comment:
since rank is already static we can check like that
```suggestion
    NODE_VALIDATION_CHECK(op, im_data_shape.size() == 4, "Im_data rank must be equal to 4.");
```

Reply:
resolved in f7d415276b3f8a48e0b32b1bd918a86d70bf6623